File: engines/java/src/main/java/io/github/davecookvectorlabs/windowcleaning/ServiceType.java
package io.github.davecookvectorlabs.windowcleaning;

/**
 * Service type — interior only, exterior only, or both.
 * The multiplier reflects the relative labour share of each service scope.
 */
public enum ServiceType {
    INTERIOR_ONLY(0.45),
    EXTERIOR_ONLY(0.55),
    BOTH(1.0);

    private final double multiplier;

    ServiceType(double multiplier) {
        this.multiplier = multiplier;
    }

    public double getMultiplier() {
        return multiplier;
    }
}

File: engines/php/src/ServiceType.php
<?php

declare(strict_types=1);

namespace WindowCleaning;

enum ServiceType: string
{
    case InteriorOnly = 'interior_only';
    case ExteriorOnly = 'exterior_only';
    case Both = 'both';

    public function multiplier(): float
    {
        return match ($this) {
            self::InteriorOnly => 0.45,
            self::ExteriorOnly => 0.55,
            self::Both => 1.0,
        };
    }
}
